[{"url":"https://api.github.com/repos/owncloud/core/issues/comments/12141587","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-12141587","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":12141587,"user":{"login":"dragotin","id":1070214,"avatar_url":"https://gravatar.com/avatar/af829234b31fb915704c1c65ef48eee0?d=https%3a%2f%2fidenticons.github.com%2f3392afab7f424e10623ba95a941f4d34.png&r=x","gravatar_id":"af829234b31fb915704c1c65ef48eee0","url":"https://api.github.com/users/dragotin","html_url":"https://github.com/dragotin","followers_url":"https://api.github.com/users/dragotin/followers","following_url":"https://api.github.com/users/dragotin/following{/other_user}","gists_url":"https://api.github.com/users/dragotin/gists{/gist_id}","starred_url":"https://api.github.com/users/dragotin/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/dragotin/subscriptions","organizations_url":"https://api.github.com/users/dragotin/orgs","repos_url":"https://api.github.com/users/dragotin/repos","events_url":"https://api.github.com/users/dragotin/events{/privacy}","received_events_url":"https://api.github.com/users/dragotin/received_events","type":"user","site_admin":false},"created_at":"2013-01-11t12:05:43z","updated_at":"2013-01-11t12:05:43z","body":"maybe its a 2gb file size limit problem, however this test was done in a 64bit environment which should be capable. probably the way to fix it is to not allow gets of files >2gb by responding with a 413 \"request entity too large\"  and have the clients using ranged downloads."},{"url":"https://api.github.com/repos/owncloud/core/issues/comments/22100673","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-22100673","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":22100673,"user":{"login":"jancborchardt","id":925062,"avatar_url":"https://gravatar.com/avatar/2fd3f4d5d762955e5b603794a888fa97?d=https%3a%2f%2fidenticons.github.com%2f84ec51f9ceed468217d4e13abf78da9c.png&r=x","gravatar_id":"2fd3f4d5d762955e5b603794a888fa97","url":"https://api.github.com/users/jancborchardt","html_url":"https://github.com/jancborchardt","followers_url":"https://api.github.com/users/jancborchardt/followers","following_url":"https://api.github.com/users/jancborchardt/following{/other_user}","gists_url":"https://api.github.com/users/jancborchardt/gists{/gist_id}","starred_url":"https://api.github.com/users/jancborchardt/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/jancborchardt/subscriptions","organizations_url":"https://api.github.com/users/jancborchardt/orgs","repos_url":"https://api.github.com/users/jancborchardt/repos","events_url":"https://api.github.com/users/jancborchardt/events{/privacy}","received_events_url":"https://api.github.com/users/jancborchardt/received_events","type":"user","site_admin":false},"created_at":"2013-08-05t11:38:10z","updated_at":"2013-08-05t11:38:10z","body":"iâ€™m closing this bug because it has been inactive for a few months. please reopen if the error still persists with owncloud 5.0.9.\r\n\r\nthanks homie!"},{"url":"https://api.github.com/repos/owncloud/core/issues/comments/23776595","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-23776595","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":23776595,"user":{"login":"steffen-moser","id":4097996,"avatar_url":"https://gravatar.com/avatar/c678832493adfc7ca3522bbf23420e29?d=https%3a%2f%2fidenticons.github.com%2f25062276810dd594285f6b99868cd649.png&r=x","gravatar_id":"c678832493adfc7ca3522bbf23420e29","url":"https://api.github.com/users/steffen-moser","html_url":"https://github.com/steffen-moser","followers_url":"https://api.github.com/users/steffen-moser/followers","following_url":"https://api.github.com/users/steffen-moser/following{/other_user}","gists_url":"https://api.github.com/users/steffen-moser/gists{/gist_id}","starred_url":"https://api.github.com/users/steffen-moser/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/steffen-moser/subscriptions","organizations_url":"https://api.github.com/users/steffen-moser/orgs","repos_url":"https://api.github.com/users/steffen-moser/repos","events_url":"https://api.github.com/users/steffen-moser/events{/privacy}","received_events_url":"https://api.github.com/users/steffen-moser/received_events","type":"user","site_admin":false},"created_at":"2013-09-04t09:28:14z","updated_at":"2013-09-04t11:23:58z","body":"i am currently experiencing the problem that downloading files larger than 2 gb fails when using the webdav access (via \"https://<hostname>/owncloud/remote.php/webdav/\") or the sync client. in the case of the sync client, both, version 1.3.0 and the recent beta version seem to be affected. \r\n\r\nthe owncloud server is running the most recent version (5.0.10) on a pure 64 bit test system (ubuntu 13.04, 64 bit). i've made the same experience with our production server which runs the same owncloud version on solaris 11.1 (x86_64) with self-compiled php 5.4.18 to have a pure 64 bit chain of apache, php and mysql (solaris 11.1 x86_64 itself still comes only with 32 bit packages of php).\r\n\r\nthe problem shows up in the following way: while uploading and downloading large files (i tested it up to 16 gb per file) via owncloud's web frontend using firefox works flawlessly on both servers (solaris production server and ubuntu test server), there still seems to be a limit at 2 gb as soon as the webdav access or the sync clients are getting involved. files larger than 2 gb are corrupted, downloaded partially or the webdav connection freezes. while it's freezing, owncloud copies the large file partially to the \"/tmp\" folder but the webdav client won't receive anything. at least from my perspective, there are still problems with handling files larger than 2 gb on pure 64 bit systems via webdav or via the sync client which need in-depth debugging. \r\n\r\ntherefore i'd suggest to re-open this bug.\r\n\r\ncan anybody else confirm this also with the recent version of owncloud? do i do anything wrong? is it working flawlessly for other users?"},{"url":"https://api.github.com/repos/owncloud/core/issues/comments/24141705","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-24141705","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":24141705,"user":{"login":"steffen-moser","id":4097996,"avatar_url":"https://gravatar.com/avatar/c678832493adfc7ca3522bbf23420e29?d=https%3a%2f%2fidenticons.github.com%2f25062276810dd594285f6b99868cd649.png&r=x","gravatar_id":"c678832493adfc7ca3522bbf23420e29","url":"https://api.github.com/users/steffen-moser","html_url":"https://github.com/steffen-moser","followers_url":"https://api.github.com/users/steffen-moser/followers","following_url":"https://api.github.com/users/steffen-moser/following{/other_user}","gists_url":"https://api.github.com/users/steffen-moser/gists{/gist_id}","starred_url":"https://api.github.com/users/steffen-moser/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/steffen-moser/subscriptions","organizations_url":"https://api.github.com/users/steffen-moser/orgs","repos_url":"https://api.github.com/users/steffen-moser/repos","events_url":"https://api.github.com/users/steffen-moser/events{/privacy}","received_events_url":"https://api.github.com/users/steffen-moser/received_events","type":"user","site_admin":false},"created_at":"2013-09-10t08:27:46z","updated_at":"2013-09-10t08:30:16z","body":"owncloud-5.0.11rc1 and sync client 1.4.0 are also affected. \r\n\r\nuploading large files (tested up to 16 gb) via web, via webdav and via sync works. downloading them works only via web, but not with using webdav or the sync client. i've been looking into this and found that the sync client really uses the webdav access. so webdav is the common source for the problem. \r\n\r\nafter spending some more time into debugging, i've found that the following patch seems to fix it for me:\r\n\r\n```php\r\n--- owncloud/3rdparty/sabre/http/response.php   2013-08-28 14:18:14.000000000 +0200\r\n+++ new/3rdparty/sabre/http/response.php        2013-09-10 09:56:32.674236315 +0200\r\n@@ -159,7 +159,10 @@ class sabre_http_response {\r\n \r\n         if (is_resource($body)) {\r\n \r\n-            fpassthru($body);\r\n+            while (!feof($body)) {\r\n+                echo fread($body, 8192);\r\n+            }\r\n+            fclose($body);\r\n \r\n         } else {\r\n```\r\n\r\ni am not an expert in php programming and i've only tested it on our solaris 11.1 server (running apache/2.2.24 (64 bit), php-5.4.18 (64 bit), mysql 5.1.37 (64 bit)) so far. it has been particularily not tested for any regressions on 32 bit servers, yet.  \r\n\r\nso maybe one of the owncloud developers could have a look into this. "},{"url":"https://api.github.com/repos/owncloud/core/issues/comments/24143809","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-24143809","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":24143809,"user":{"login":"deepdiver1975","id":1005065,"avatar_url":"https://gravatar.com/avatar/bdba071ed68da9ce8edae53c364902ef?d=https%3a%2f%2fidenticons.github.com%2f67e101ad9531b7d25697b76e503e062e.png&r=x","gravatar_id":"bdba071ed68da9ce8edae53c364902ef","url":"https://api.github.com/users/deepdiver1975","html_url":"https://github.com/deepdiver1975","followers_url":"https://api.github.com/users/deepdiver1975/followers","following_url":"https://api.github.com/users/deepdiver1975/following{/other_user}","gists_url":"https://api.github.com/users/deepdiver1975/gists{/gist_id}","starred_url":"https://api.github.com/users/deepdiver1975/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/deepdiver1975/subscriptions","organizations_url":"https://api.github.com/users/deepdiver1975/orgs","repos_url":"https://api.github.com/users/deepdiver1975/repos","events_url":"https://api.github.com/users/deepdiver1975/events{/privacy}","received_events_url":"https://api.github.com/users/deepdiver1975/received_events","type":"user","site_admin":false},"created_at":"2013-09-10t09:08:19z","updated_at":"2013-09-10t09:08:19z","body":"actually this is related to sabredav @evert can i ask you for your opinion? thx"},{"url":"https://api.github.com/repos/owncloud/core/issues/comments/24151698","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-24151698","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":24151698,"user":{"login":"evert","id":178960,"avatar_url":"https://gravatar.com/avatar/7d76e2bb6f8c962a5628093c9f5bc6fb?d=https%3a%2f%2fidenticons.github.com%2f46192a8c94cd97601476c800f3d5f483.png&r=x","gravatar_id":"7d76e2bb6f8c962a5628093c9f5bc6fb","url":"https://api.github.com/users/evert","html_url":"https://github.com/evert","followers_url":"https://api.github.com/users/evert/followers","following_url":"https://api.github.com/users/evert/following{/other_user}","gists_url":"https://api.github.com/users/evert/gists{/gist_id}","starred_url":"https://api.github.com/users/evert/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/evert/subscriptions","organizations_url":"https://api.github.com/users/evert/orgs","repos_url":"https://api.github.com/users/evert/repos","events_url":"https://api.github.com/users/evert/events{/privacy}","received_events_url":"https://api.github.com/users/evert/received_events","type":"user","site_admin":false},"created_at":"2013-09-10t11:05:22z","updated_at":"2013-09-10t11:05:22z","body":"fpassthru should just work... so if it doesn't, it would be good to somehow get a reproducable test case for this"},{"url":"https://api.github.com/repos/owncloud/core/issues/comments/25391545","html_url":"https://github.com/owncloud/core/issues/1149#issuecomment-25391545","issue_url":"https://api.github.com/repos/owncloud/core/issues/1149","id":25391545,"user":{"login":"steffen-moser","id":4097996,"avatar_url":"https://gravatar.com/avatar/c678832493adfc7ca3522bbf23420e29?d=https%3a%2f%2fidenticons.github.com%2f25062276810dd594285f6b99868cd649.png&r=x","gravatar_id":"c678832493adfc7ca3522bbf23420e29","url":"https://api.github.com/users/steffen-moser","html_url":"https://github.com/steffen-moser","followers_url":"https://api.github.com/users/steffen-moser/followers","following_url":"https://api.github.com/users/steffen-moser/following{/other_user}","gists_url":"https://api.github.com/users/steffen-moser/gists{/gist_id}","starred_url":"https://api.github.com/users/steffen-moser/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/steffen-moser/subscriptions","organizations_url":"https://api.github.com/users/steffen-moser/orgs","repos_url":"https://api.github.com/users/steffen-moser/repos","events_url":"https://api.github.com/users/steffen-moser/events{/privacy}","received_events_url":"https://api.github.com/users/steffen-moser/received_events","type":"user","site_admin":false},"created_at":"2013-09-30t18:45:15z","updated_at":"2013-09-30t18:45:15z","body":"i think i can provide you detailed step-by-step instructions to reproduce this behavior.\r\n\r\nlet me summarize the problem:\r\n\r\non a pure 64 bit server system (linux ubuntu 13.04 x86_64, oracle solaris 11.1 x86_64) downloading files larger than 2 gb can fail (download stops, files get corrupted) when downloading the files by using owncloud's sabredav access (e.g. via web browser, via webdav mount or via the sync client). uploading and downloading of the large files doesn't cause problems when doing that via owncloud's web interface.\r\n\r\n\r\ndetailed steps to reproduce the problem:\r\n\r\n1. do a base installation of a recent linux distribution (e.g. ubuntu 13.04), choose, of course, the **x86_64** edition.\r\n\r\n2. install the apache web browser that comes with the distribution, but **don't install** the php version which is packaged by the distributor. i installed the following packages after a standard installation of ubuntu (and all of its dependencies):\r\n\r\napache2-mpm-prefork \r\napache2-prefork-dev\r\nlibxml2-dev \r\nlibpng12-dev\r\n\r\n3. download a recent version of php (vanilla sources) from php.net. i've chosen version 5.4.20 for this set-up, but 5.5.4 shows the same problem.\r\n\r\n4. untar php, configure it, compile it (make) and install it (sudo make installi). i configured it with the following options:\r\n\r\nconfigure --with-zlib --with-gd --enable-mbstring --enable-zip --with-pear --with-mysql --with-apxs2=/usr/bin/apxs2\r\n\r\n5. configure a handler in apache for php in \"/etc/apache2/sites-enabled/000-default/\":\r\n\r\naddhandler php5-script .php\r\n\r\n6. change the allowoverride option in the same file from \"none\" to \"all\"\r\n\r\n7. restart apache\r\n\r\n8. download owncloud 5.0.11 (don't install the version that has been packaged by the linux distributor) and extract the tar file to \"/var/www/owncloud\". give the web server user (www-data) write access to the necessary directories.\r\n\r\n9. access owncloud with a web browser, create the admin account and change the max. file size to 20 gb. \r\n\r\n10. create some test files (e.g. using \"dd\") with sizes, for example, of 512 mb, 1 gb, 2 gb, 3 gb, 4 gb, ..., 8 gb. \r\n\r\n11. upload these files via owncloud's web frontend. uploading these files should work without any problems. after then download the files via ownclouds's web frontend. this should also work without problems and the downloaded files will not be corrupted (you can compare them to the original ones).\r\n\r\n12. now try to download the files again, but this time by using the sabredav client which comes with owncloud. therefore enter the following url: http://<hostname>/owncloud/remote.php/webdav\r\n\r\nthe files with 512 mb and 1 gb will be downloadable without any problems. the files with 2 gb and 3 gb won't be downloadable (connection reset by apache), files larger than 4 gb will be downloadable but the downloaded files will be corrupted (by being downloaded only partially). as the owncloud sync client also uses the sabredav interface, it'll run into the same problem. the problem also occurs if someone mounts the sabredav url in the operating system.\r\n\r\nthe problem does not occur if you use a version of php which has been treated with the patch which resulted from these bug reports:\r\n\r\nhttps://bugs.php.net/bug.php?id=52102\r\nhttps://bugs.launchpad.net/ubuntu/+source/php5/+bug/564920\r\n\r\n\r\nthis patch is included in recent linux distributions (e.g. ubuntu 13.04). therefore, the problem does not occur to the majority of owncloud users as long as they use patched versions of php. we run owncloud on a oracle solaris 11.1 server (on a x86_64 platform) as a host for owncloud and other web applications. while solaris 11.1 has a 64 bit kernel and comes with 64 bit versions of apache and mysql, it does only contain 32 bit packages of php (and the rather old version 5.3.14). to enable the handling of very large files with owncloud, we built ips packages for solaris 11.1 containing 64 bit binaries of the latest php 5.4.x by ourselves. after installing these packages, handling large files via owncloud's web frontend has started to work like a charm, but the sync client and webdav have still been making trouble.\r\n\r\npossible solutions: when i apply the above given patch to \"owncloud/3rdparty/sabre/http/response.php\", the problem will be fixed (at least it seems to be fixed to me). alternative solution for us: we patch php by ourselves like linux distributors do. \r\n\r\nin my opinion, owncloud should be able to handle very large files at least in a pure 64 bit environment without the need of linux distributors' patches, but also with a vanilla php interpreter."}]